internal/report/report.go

```diff
@@ -1,383 +1,190 @@
-func getPolicyName(policy policiesv1.Policy) (string, error) {
-	switch policy.GetObjectKind().GroupVersionKind() {
-	case schema.GroupVersionKind{
-		Group:   constants.KubewardenPoliciesGroup,
-		Version: constants.KubewardenPoliciesVersion,
-		Kind:    constants.KubewardenKindClusterAdmissionPolicy,
-	}:
-		return "cap-" + policy.GetName(), nil
-	case schema.GroupVersionKind{
-		Group:   constants.KubewardenPoliciesGroup,
-		Version: constants.KubewardenPoliciesVersion,
-		Kind:    constants.KubewardenKindAdmissionPolicy,
-	}:
-		return "ap-" + policy.GetName(), nil
-	default:
-		// this should never happen
-		log.Fatal().Msg("cannot generate policy name")
-		return "", fmt.Errorf("cannot generate policy name")
+	if allowed {
+		return statusPass
 	}
-}
 
-func getResourceObjectReference(resource unstructured.Unstructured) v1.ObjectReference {
-	return v1.ObjectReference{
-		Kind:            resource.GetKind(),
-		Namespace:       resource.GetNamespace(),
-		Name:            resource.GetName(),
-		UID:             resource.GetUID(),
-		APIVersion:      resource.GetAPIVersion(),
-		ResourceVersion: resource.GetResourceVersion(),
-	}
+	return statusFail
 }
 
-//nolint:funlen
-func newPolicyReportResult(
-	policy policiesv1.Policy, resource unstructured.Unstructured,
-	auditResponse *admv1.AdmissionReview, responseErr error,
-) *v1alpha2.PolicyReportResult {
-	var result v1alpha2.PolicyResult
-	var description string
-	if responseErr != nil {
-		result = StatusError
-		description = auditResponse.Response.Result.Message
-	} else {
-		if auditResponse.Response.Allowed {
-			result = StatusPass
-		} else {
-			result = StatusFail
-			description = auditResponse.Response.Result.Message
-		}
-	}
+func computePolicyResultSeverity(policy policiesv1.Policy) wgpolicy.PolicyResultSeverity {
+	var severity wgpolicy.PolicyResultSeverity
 
-	name, _ := getPolicyName(policy)
-
-	time := metav1.Now()
-	timestamp := *time.ProtoTime()
-
-	var severity v1alpha2.PolicyResultSeverity
-	var scored bool
 	if policy.GetPolicyMode() == policiesv1.PolicyMode(policiesv1.PolicyModeStatusMonitor) {
-		scored = true
-		severity = SeverityInfo
-	} else {
-		if sev, present := policy.GetSeverity(); present {
-			scored = true
-			switch sev {
-			case SeverityCritical:
-				severity = SeverityCritical
-			case SeverityHigh:
-				severity = SeverityHigh
-			case SeverityMedium:
-				severity = SeverityMedium
-			case SeverityLow:
-				severity = SeverityLow
-			default:
-				// this should never happen
-				log.Error().
-					Dict("result", zerolog.Dict().
-						Str("policy", policy.GetName()).
-						Str("resource", resource.GetName()).
-						Bool("allowed", auditResponse.Response.Allowed).
-						Str("severity", sev),
-					).Msg("severity unknown")
-			}
-		}
+		return severityInfo
 	}
 
-	var category string
-	if cat, present := policy.GetCategory(); present {
-		category = cat
+	if s, present := policy.GetSeverity(); present {
+		return wgpolicy.PolicyResultSeverity(s)
```

Comment:
Our controller does not validate the severity set by the user. Therefore, I'm wondering if the user set an invalid value, audit scanner can fail somehow. I don't think we need to change here. But it's something to remember. ;)

Reply:
we will just output an empty severity